google/resource_spanner_database.go

```diff
@@ -0,0 +1,239 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/spanner/v1"
+)
+
+func resourceSpannerDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSpannerDatabaseCreate,
+		Read:   resourceSpannerDatabaseRead,
+		Delete: resourceSpannerDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceSpannerDatabaseImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"instance": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+
+					if len(value) < 2 && len(value) > 30 {
+						errors = append(errors, fmt.Errorf(
+							"%q must be between 2 and 30 characters in length", k))
+					}
+					if !regexp.MustCompile("^[a-z0-9-]+$").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q can only contain lowercase letters, numbers and hyphens", k))
+					}
+					if !regexp.MustCompile("^[a-z]").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q must start with a letter", k))
+					}
+					if !regexp.MustCompile("[a-z0-9]$").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q must end with a number or a letter", k))
+					}
+					return
+				},
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"ddl": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
```

Comment:
There is an update method for this. If you decide not to implement it in this PR, please file an issue and add a TODO here link to the issue.